pkg/controllers/user/globaldns/user_cluster_globaldns_controller.go

```diff
@@ -56,67 +77,130 @@ func (g *UserGlobalDNSController) sync(key string, obj *v3.GlobalDNS) (runtime.O
 		return nil, err
 	}
 
-	// go through target projects which are part of the current cluster
+	// go through target projects which are part of the current cluster and find all ingresses
+	var allIngresses []*v1beta1.Ingress
+
 	for _, t := range mcapp.Spec.Targets {
 		split := strings.SplitN(t.ProjectName, ":", 2)
 		if len(split) != 2 {
-			return mcapp, fmt.Errorf("error in splitting project ID %v", t.ProjectName)
+			return nil, fmt.Errorf("error in splitting project ID %v", t.ProjectName)
 		}
 		// check if the target project in this iteration is same as the cluster in current context
 		if split[0] != g.clusterName {
 			logrus.Debugf("Continuing since target is not for current cluster since project %v doesn't belong in cluster %v", split[1], g.clusterName)
```

Comment:
I wonder if we need to log this even in debug, because this is just our logic and won't help in debugging purposes. Also would always happen for all cluster controllers

Reply:
removed the log